### DISCUSSION TITLE
Stock sublocations display

### DISCUSSION THREAD

**Discussion by sintech** [inventree/InvenTree#4549] [General]
Hello, a small feedback if you don't mind.
Recently I updated inventree to latest release and noticed that Stock locations are displayed in a new way.
Every location has "Load sublocation" button which does not make any sense. Because It implies that I must click all of them to see whether it has sublocation or not. In previous versions Inventree just displayed full tree view of all my locations with item count and I could easily select the needed one.

In my understanding it was made to speedup loading time, but it is clearly a degradation in usability and clarity. The same could be said about quick category/location selector at the top of the page, it just not as good as left-side tree view from early versions.

![изображение](https://user-images.githubusercontent.com/10057327/229279485-5bcec5f0-cfff-49e4-b184-aa98dc727b51.png)

P.S. Thank you for your continued work on Inventree. Please don't forget about basics while doing new cool things.

**Comment by SchrodingersGat** [comment]
@sintech thanks for the feedback, UX is always hard and in the last few dev cycles we have been focusing on moving data access to the API and improving query efficiency.

This issue seems related to https://github.com/inventree/InvenTree/issues/3814 - and has been discussed before elsewhere also. If you have any ability to contribute an improved solution here that would be greatly appreciated.

**Comment by wolflu05** [comment]
You can also disable this lazy loading feature via the settings, however if you have big trees, this will result in bad performance.

**Reply by sintech** [reply]
Thank you, I checked all setting again and found this option in "Server Settings" to specify tree loading depth.

**Comment by matmair** [comment]
You are welcome to provide PRs that remove the `degradation of usability` without locking up the whole UI on large instances - which I see as a much larger problem.

**Comment by sintech** [comment]
Solution is found, it is possible to specify tree depth to display by default. I was sure that it must be configurable but looked in the wrong place.
Anyway "Load sublocation" button should be displayed only for locations that have sublocations. It looks like an easy fix.
